src/kronecker_array.hpp

```diff
@@ -125,7 +125,7 @@ class kronecker_array
     static limit_type determine_limit(const size_type &m)
     {
         piranha_assert(m >= 1u);
-        std::mt19937 engine(static_cast<unsigned long>(m));
+        std::mt19937 engine(static_cast<unsigned>(m));
```

Comment:
@isuruf Good catch! Would you mind casting to ``std::mt19937::result_type`` instead?